common/client/transaction_sender.go

```diff
@@ -0,0 +1,245 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+type TransactionSender[TX any] interface {
+	SendTransaction(ctx context.Context, tx TX) (SendTxReturnCode, error)
+	Close() error
+}
+
+// TxErrorClassifier - defines interface of a function that transforms raw RPC error into the SendTxReturnCode enum
+// (e.g. Successful, Fatal, Retryable, etc.)
+type TxErrorClassifier[TX any] func(tx TX, err error) SendTxReturnCode
+
+type sendTxResult struct {
+	Err        error
+	ResultCode SendTxReturnCode
+}
+
+const sendTxQuorum = 0.7
+
+// SendTxRPCClient - defines interface of an RPC used by TransactionSender to broadcast transaction
+type SendTxRPCClient[TX any] interface {
+	// SendTransaction errors returned should include name or other unique identifier of the RPC
+	SendTransaction(ctx context.Context, tx TX) error
+}
+
+func NewTransactionSender[TX any, CHAIN_ID types.ID, RPC SendTxRPCClient[TX]](
+	lggr logger.Logger,
+	chainID CHAIN_ID,
+	chainFamily string,
+	multiNode *MultiNode[CHAIN_ID, RPC],
```

Comment:
We can make TransactionSender a public struct since it's just the single implementation. Same with MultiNode.

Do you think it's necessary to create an interface for MultiNode here to be accepted rather than using the struct?